### PR TITLE
Refactor Project Structure for Modding

### LINKED ISSUE #191: Put world editor content into `user://` and allow loading from there
The data should be located in `user://tracks` or `user://editor`. Additionally direct loading of the project from that file location needs to be supported.

### PR DESCRIPTION
**Pull request by DasCapschen**
Fixes #191.

Huge changes to project structure. 
Editor and Game Scripts have been separated (at least in folders, not in logic)

No more "Basic" Resources, just Resources!

No more `addons` folder for game files. (jTools remains in there)

Added a "modding tools" addon that allows creating a new mod.
Should be a standalone Godot addon, but I cba to make a new repo for it.

Also adds a "content.tres".
This lists all the folders that contain content, like objects, materials, sounds, ... 
The Editor can then load these folders so the user can select content from it.

Content.tres also contains a list of trains and worlds, so the main menu can show them in the list.

Content.tres also contains things like mod name, mod authors (uses the Authors Resource), mod version, a list of other mods this mod depends on. So this will be used for loading and displaying mods.

### CODE REVIEW

**Comment by DasCapschen** [comment]
Please don't merge anything before this is done... I cannot rebase this...

Currently does not work. I still stumble across errors where wrong paths are used. Somehow Godots moving/renaming isn't as robust as you'd think...

**Comment by DasCapschen** [comment]
Game should work just like before. 
Editor is currently broken, because it's still looking for resources in `res://Resources`, but in the coming days I will change it so it looks stuff up in the content.tres.
Also loaded .pcks will currently not show up in editor, I will also change the pck loading to append their content.tres info to the global content "repo"

**Comment by HaSa1002** [comment]
Sounds good. I see the repaired content loading as merge requirement. I don't want to repair the main branch again.

**Comment by HaSa1002** [comment]
grafik

**Comment by HaSa1002** [comment]
My whole resource folder vanished

**Comment by HaSa1002** [comment]
Our Resources went 💥

**Comment by DasCapschen** [comment]
.gitignore caused that 🙈 
```
 src/Resources/*
 !src/Resources/Tutorials
 !src/Resources/Basic
 ```

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,3 @@
...
-
-src/Resources/*
-!src/Resources/Tutorials
-!src/Resources/Basic
-
-src/Trains/*
-!src/Trains/JFR1/
-
 src/.import/
 *.csv.import
```
Comment (HaSa1002) [inline .gitignore:10]: This feels wrong

**Review by HaSa1002 — APPROVED** [review]
Driving works.
Editing works as well. (Some issues exist, but are independet of this one).
Please rebase (Most importantly the resources removal and addition to get rid of 200MB of History).
Please update https://github.com/Libre-TrainSim/Libre-TrainSim/wiki/How-to-update-a-track-from-v0.8-to-v0.9 on merge.

*This is a blind approval. It is a required change and we need to merge it ASAP. Regression need to be fixed in subsequent PRs*

**NOTE: This PR breaks your world saves.**

**Comment by DasCapschen** [comment]
Information on how to update worlds will be added to the 0.8 to 0.9 transition guide.  
Worlds need to be reworked for 0.9 anyways.

**Pull request merged**